model_analyzer/record/metrics_manager.py

```diff
@@ -166,21 +166,22 @@ def execute_run_config(self, run_config):
         measurement to the result manager
         """
 
-        # Create model variant
-        self._create_model_variant(original_name=run_config.model_name(),
-                                   variant_config=run_config.model_config())
+        # Create model variants
+        for model_config in run_config.model_configs():
+            self._create_model_variant(original_name=run_config.model_name(),
```

Comment:
is this function smart enough to do nothing if it is already done? Unlike before, much of the time the model variant will already exist from a previous run.

Reply:
yes, if it fails to create the directory, it just continues with the rest of the program:

https://github.com/triton-inference-server/model_analyzer/blob/main/model_analyzer/record/metrics_manager.py#L223-L224

Reply:
What about the 'first config variant' code? That seems like it wouldn't work for multiple models?

Reply:
updated logic to have a first config variant per model 👍

Reply:
looks like it should work. Is there any testing or at least a manual run to confirm?

Reply:
This can't really be tested yet with multiple models since other end-to-end pieces are missing, and it's not well set up for a unit test (an L0 is better since the feature we want to test ranges across several layers of the code execution, meaning the mocking would be a lot).

I did test, however, with one model, and the old behavior still works. The variant directories get created correctly and all subsequent variant directories still link to the first variant.